Add orbit<>bold integration article (#1413)
OffchainLabs/arbitrum-docs: arbitrum-docs/launch-orbit-chain/bold-adoption-for-orbit-chains.mdx

---
title: 'BoLD for Orbit chains'
description: ''
user_story: 'As a developer or researcher of the Arbitrum product suite, I want to learn about BoLD, a next-generation dispute protocol that enables permissionless validation.'
content_type: release announcement
author: leeederek
sme: leeederek
target_audience: 'Developers who want to build on Arbitrum using popular programming languages, like Rust'
sidebar_position: 1
---

:::info PUBLIC PREVIEW DOCUMENT

This document is currently in **public preview** and may change significantly as feedback is captured from readers like you. Click the _Request an update_ button at the top of this document or [join the Arbitrum Discord](https://discord.gg/arbitrum) to share your feedback.

:::

## Launch details and key dates

- **Status:** Alpha - continued testing and evaluation of performance of BoLD with Stylus

> - **Sepolia** Mid August 2024                                             

Reviewer: ```suggestion
- **Arbitrum Sepolia** Q4 2024                                           
```

> - <a data-quicklook-from="arbitrum-one">**Arbitrum One**</a> Late Sept 2024

Reviewer: ```suggestion
- <a data-quicklook-from="arbitrum-one">**Arbitrum One**</a> Q4 2024
```

> - <a data-quicklook-from="arbitrum-nova">**Arbitrum Nova**</a> Late Sept 2024

Reviewer: ```suggestion
- <a data-quicklook-from="arbitrum-nova">**Arbitrum Nova**</a> Q4 2024
```

### tldr;

Arbitrum BoLD is an upgrade to the existing dispute protocol used by Arbitrum chains that enables permissionless validation and other security benefits. Similar to other new features for the Arbitrum technology stack, BoLD will be made available for mainnet adoption by Orbit chains when they are ready. Specifically for Arbitrum BoLD, an upgrade will involve a Nitro node software version bump and the deployment and consumption of new smart contracts on the parent chain.

Adopting <a data-quicklook-from="bold">Arbitrum BoLD</a> brings new security benefits to Orbit chains that are independent of whether or not validation is permissionless. These new security benefits include improved resistance to delay attacks and increased censorship resistance for `L3s`. Because of these new security benefits, we strongly recommend teams to upgrade their Orbit chains to use Arbitrum BoLD (for the security benefits) but **keep validation permissioned**. Below is a quick breakdown of the impacts of permissioned BoLD vs. permissionless BoLD on your Orbit chain:

> #### Permissioned BoLD on an Orbit chain

Reviewer: @anegg0 - i still think there are better ways to present this information (compared to 2 lists you have now). Could we perhaps take inspiration from something like these two examples below?

<img width="1353" alt="image" src="https://github.com/user-attachments/assets/a1985614-2f08-41c6-9185-949b2eb6553a">
<img width="903" alt="image" src="https://github.com/user-attachments/assets/b122099c-7715-4a8b-a258-6623532ee695">

Author: I uploaded a new iteration of the diagram, wdyt?

##### Benefits                                                                                                           
    - Improved resistance to delay attacks                                                                               
    - Increased censorship resistance for L3s                                                                            
    - Use of any `ERC-20` as the bonding token                                                                           
    - Be on the latest iteration of Arbitrum technology, making future security & feature updates much easier to pick up 

>     - Less engineering overhead and infrastructure costs                                                                 

Reviewer 2: i would think the overhead and costs are the same pre and post bold

##### Caveats 

>     - Permissioned validation: only entities on allowlist can resolve disputes

Reviewer 2: might add as a benefit that you get 1/N honest assumption

Reviewer: Makes sense - though this is true even without BoLD. I will add it though.

#### Permissionless BoLD on an Orbit chain

##### Benefits                                                                                                           
    - Improved resistance to delay attacks                                                                               
    - Increased censorship resistance for L3s                                                                            
    - Use of any `ERC-20` as the bonding token                                                                           
    - Be on the latest iteration of Arbitrum technology, making future security & feature updates much easier to pick up 

>     - Less engineering overhead and infrastructure costs                                                                 
>     - More trust in the system's integrity and fairness                                                                  
>     - Creates a more open ecosystem for third-party developers and users                                                 

Reviewer 2: ```suggestion
    - Anyone can participate in validation                                               
```

Reviewer: Added!

##### Caveats 

>     - Resource exhaustion attacks: challenges require computing and financial resources to participate in, and an attacker could have significantly more resources than the defenders

Reviewer 2: ```suggestion
    - Resource exhaustion attacks: challenges require computing and financial resources to participate in. An attacker that has significantly more resources than the defenders could compromise safety of the chain.
```

Reviewer: added!

>     - Liveness impact: each challenge takes time to resolve, and so if the cost of a challenge is cheap, then an attacker can open challenges and prevent the progress of the chain at a low price 

Reviewer 2: ```suggestion
    - Liveness impact: if the cost of a challenge is cheap, then an attacker can cheaply open a challenge and prevent the progress of the chain for as long as they can censor honest moves
```

censorship of honest moves is also required to delay progress

Reviewer: Added!

>     - Liquidity and price impacts: if a low-liquidity (relative) `ERC-20` token is used as the bonding token and a challenge occurs                                                                                        ### Benefits of adopting Arbitrum BoLD

Reviewer 2: ```suggestion
    - Liquidity and price impacts: if a low-liquidity (relative) `ERC-20` token is used as the bonding token and a challenge occurs
```

Reviewer: Good catch

    - Grieving attacks: if the reward to defenders incentivizes self-challenges (by being too high), effectively lowering the cost of attack
    - More expensive engineering overhead and infrastructure costs

> 

Reviewer: ```suggestion
### Benefits of Adopting Arbitrum BoLD for your Orbit chain
```

#### Being on the latest version of Arbitrum technology
Adopting Arbitrum BoLD for your Orbit chain will require upgrading the Nitro node software and deploying a new set of contracts on your parent chain. While not specifically related to Arbitrum BoLD, it is always strongly recommended that Orbit chain owners upgrade and keep their chain on the latest stable releases of both Nitro node software and the relevant on-chain contracts. This is critical to ensure your Orbit chain benefits from the latest security improvements and features that the Offchain Labs team is constantly churning out.

### Caveats that come with adopting Arbitrum BoLD for permissionless validation

Arbitrum BoLD's implementation and specification have been thoroughly tested and audited. The upgrade to Arbitrum BoLD is not the subject of this section, but rather the caveats and nuances that come with whether to enable permissionless validation. 

:::warning

It is strongly recommended that existing and prospective Orbit chains upgrade to use Arbitrum BoLD but **keep validation permissioned**.

:::

> Enabling permissionless validation means that any entity can spin up a validator and open challenges to dispute invalid claims made by other validators on the network. This opens up an Orbit chain to the risk of spam and attacks by unknown and malicious entities. To mitigate this risk for Arbitrum One, a considerable amount of research and testing has been done to optimize the trade-offs between deterring attacks and managing the costs of defending Arbitrum for honest parties. This research includes carefully calculating all relevant bond sizes, challenge period durations, and relevant plans for operating the infrastructure. More information on this research can be found in the [BoLD whitepaper](https://arxiv.org/abs/2404.10491). Below are a few examples of various risks that an Orbit chain will hold should they pursue permissionless BoLD:

Reviewer: ```suggestion
Enabling permissionless validation means that any entity can spin up a validator and open challenges to dispute invalid claims made by other validators on the network. This opens up an Orbit chain to the risk of spam and attacks by unknown and malicious entities. To mitigate this risk for Arbitrum One, a considerable amount of research and testing has been done to optimize the trade-offs between deterring attacks and managing the costs of defending Arbitrum for honest parties. This research includes carefully calculating all relevant bond sizes, challenge period durations, and relevant plans for operating the infrastructure. More information on this research can be found in the [BoLD whitepaper](https://arxiv.org/abs/2404.10491). Lastly, the elected members of the [Arbitrum Security Council](https://docs.arbitrum.foundation/concepts/security-council) may intervene should any of these risks manifest and would otherwise harm the ArbitrumDAO-owned chains and its users. Below are a few examples of various risks that an Orbit chain will hold should they pursue permissionless BoLD:
```

#### Risk of resource exhaustion attacks
Where malicious entities can acquire and utilize more resources than honest parties can put together during a challenge. Such an attack can take many forms and includes both on-chain and off-chain computational/infra costs. For example, a well-coordinated attack on an Orbit chain could overwhelm honest parties if the malicious actors can spend more gas and computational power and acquire more of the bonding asset than the defenders can. This risk can be mitigated by a combination of high bond sizes, use of a price-independent bonding asset, use of a bonding asset with high liquidity, strong economic guarantees that attackers will lose most of their resources, sufficiently long challenge periods, and robust infrastructure operations and resources that can respond and scale up when necessary. More information on resource exhaustion attacks and how Arbitrum BoLD's design accounts for this risk can be found in Section 6.1.4 of the [BoLD whitepaper](https://arxiv.org/abs/2404.10491).

> #### Risks to liveness or delays of the chain
> If the bond sizes are set too low, making challenges relatively cheap to open compared to the benefits of delaying the chain. Remember that challenges, while time-bound, still take time to complete. This means that each challenge adds a fixed amount of time to confirm assertions. If the bond sizes are carefully calculated and chosen, confirmations of an Orbit chain's state may be completed on time if the cost to open a challenge is low. Delaying the confirmation of assertions for a chain could negatively impact the chain in many ways that an attacker could benefit from (e.g., profiting from price volatility and price impacts on the Orbit chain's token may make delaying the chain worthwhile for an attacker).

Reviewer 2: ```suggestion
#### Risks to liveness or delays of the chain
If the bond sizes are set too low, an adversary can cheaply create a challenge and delay confirmation of an assertion for up to an entire extra challenge period if they can censor honest BOLD moves. Remember that challenges, while time-bound, still take time to complete. Delaying the confirmation of assertions for a chain could negatively impact the chain in many ways that an attacker could benefit from (e.g., profiting from price volatility and price impacts on the Orbit chain's token may make delaying the chain worthwhile for an attacker).
```

> 

Reviewer: ```suggestion
### Conclusion for Orbit chains considering BoLD Permissionless Validation
```

---
title: 'BoLD for Orbit chains'
description: ''
user_story: 'As a developer or researcher of the Arbitrum product suite, I want to learn about BoLD, a next-generation dispute protocol that enables permissionless validation.'
content_type: release announcement
author: leeederek
sme: leeederek
target_audience: 'Developers who want to build on Arbitrum using popular programming languages, like Rust'
sidebar_position: 1
---

:::info PUBLIC PREVIEW DOCUMENT

This document is currently in **public preview** and may change significantly as feedback is captured from readers like you. Click the _Request an update_ button at the top of this document or [join the Arbitrum Discord](https://discord.gg/arbitrum) to share your feedback.

:::

## Launch details and key dates

- **Status:** Alpha - continued testing and evaluation of performance of BoLD with Stylus

> | Chain            | Release timelines                                           |

Author: Done.

| <a data-quicklook-from="arbitrum-nova">Arbitrum Nova</a>    | Late Sept 2024                                              |

### tldr;

Arbitrum BoLD is an upgrade to the existing dispute protocol used by Arbitrum chains that enables permissionless validation and other security benefits. Similar to other new features for the Arbitrum technology stack, BoLD will be made available for mainnet adoption by Orbit chains when they are ready. Specifically for Arbitrum BoLD, an upgrade will involve a Nitro node software version bump and the deployment and consumption of new smart contracts on the parent chain.

Adopting <a data-quicklook-from="bold">Arbitrum BoLD</a> brings new security benefits to Orbit chains that are independent of whether or not validation is permissionless. These new security benefits include improved resistance to delay attacks and increased censorship resistance for `L3s`. Because of these new security benefits, we strongly recommend teams to upgrade their Orbit chains to use Arbitrum BoLD (for the security benefits) but **keep validation permissioned**. Below is a quick breakdown of the impacts of permissioned BoLD vs. permissionless BoLD on your Orbit chain:

> | Permissioned BoLD                           | Permissionless BoLD                              |

Author: I modified the permissioned/permissionless trade-off tables.
The format I'm proposing would only be relevant if we have solid arguments to fill in the blanks. If we don't, we could remove the table and mention the trade-offs in the text, as you did in the previous version.

lmk what you think :)

> | Permissioned BoLD                           | Permissionless BoLD                              |

Author: @leeederek , can you confirm?

> | Permissioned BoLD                           | Permissionless BoLD                              |

Reviewer: I'm resolving this comment in favor of iterating on this one (that is talking about the same issue): https://github.com/OffchainLabs/arbitrum-docs/pull/1413#discussion_r1766880803
